unit_tests/test_redis_aggregator.py

```diff
@@ -0,0 +1,334 @@
+                 return_value=TEST_TRANSACTION_ID)
+    mocker.patch("gsy_e_sdk.redis_aggregator.wait_until_timeout_blocking")
+
+
+@pytest.fixture(name="mock_client_command_buffer_attributes")
+def fixture_mock_client_command_buffer(mocker):
+    mocker.patch("gsy_e_sdk.redis_aggregator.ClientCommandBuffer.buffer_length",
+                 new_callable=PropertyMock, return_value=3)
+    mocker.patch("gsy_e_sdk.redis_aggregator.ClientCommandBuffer.execute_batch",
+                 return_value=TEST_BATCH_COMMAND_DICT)
+
+
+@pytest.fixture(name="aggregator")
+def fixture_aggregator(mocker):
+    mocker.patch("gsy_e_sdk.redis_aggregator.uuid.uuid4",
+                 return_value=TEST_TRANSACTION_ID)
+    mocker.patch("gsy_e_sdk.redis_aggregator.wait_until_timeout_blocking")
+    return RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+
+
+class TestRedisAggregatorConstructor:
+    """Include tests for the constructor of the RedisAggregator class."""
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_grid_fee_calculation_instantiated():
+        grid_fee_calculation_mock = MagicMock()
+        with patch("gsy_e_sdk.redis_aggregator.GridFeeCalculation",
+                   return_value=grid_fee_calculation_mock) as mocked_class:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            mocked_class.assert_called()
+            assert agg.grid_fee_calculation is grid_fee_calculation_mock
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_redis_db_instantiated():
+        redis_db_mock = MagicMock()
+        with patch("gsy_e_sdk.redis_aggregator.StrictRedis.from_url",
+                   return_value=redis_db_mock) as redis_from_url:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            redis_from_url.assert_called_with(LOCAL_REDIS_URL)
+            assert agg.redis_db is redis_db_mock
+
+    @staticmethod
+    def test_constructor_pubsub_instantiated(aggregator):
+        aggregator.redis_db.pubsub.assert_called()
+        assert aggregator.pubsub is aggregator.redis_db.pubsub()
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_client_command_buffer_instantiated():
+        client_command_buffer_mock = MagicMock()
+        with patch("gsy_e_sdk.redis_aggregator.ClientCommandBuffer",
+                   return_value=client_command_buffer_mock) as mocked_class:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            mocked_class.assert_called()
+        assert agg._client_command_buffer is client_command_buffer_mock
+
+    @staticmethod
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_threadpoolexecutor_instantiated():
+        ThreadPoolExecutor_mock = MagicMock
```

Comment:
Nit:
```suggestion
        thread_pool_executor_mock = MagicMock
```

Reply:
I'm not really sure how the compiler let that styling pass. I should have been warned about that...